### PR TITLE
Fix bugzilla issue 24153 - Inliner breaks -betterC by requiring TypeInfo

### PR DESCRIPTION
**Pull request by bangbangsheshotmedown**

### CODE REVIEW

File: compiler/src/dmd/expressionsem.d [hunk 1]
```diff
@@ -17184,8 +17184,11 @@ void semanticTypeInfo(Scope* sc, Type t)
             Scope scx;
             scx.eSink = global.errorSink;
             scx._module = sd.getModule();
-            getTypeInfoType(sd.loc, t, &scx);
-            sd.requestTypeInfo = true;
+            if (global.params.betterC == false)
```
Comment (thewilsonator) [inline compiler/src/dmd/expressionsem.d:17187]: please use `useTypeInfo` instead of `betterC`.

**Review by thewilsonator — APPROVED** [review]
This otherwise looks good. If you consider this issue to be a blocker, you might want to rebase to target `stable` instead of `master`.

**Review by thewilsonator — APPROVED** [review]

File: compiler/src/dmd/expressionsem.d (hunk 1 above)
Comment (bangbangsheshotmedown) [reply compiler/src/dmd/expressionsem.d]: done

**Comment by bangbangsheshotmedown** [comment]
> This otherwise looks good. If you consider this issue to be a blocker, you might want to rebase to target `stable` instead of `master`.

I don't know how to do that, the command fails: `fatal: Does not point to a valid commit 'upstream/stable'`

**Comment by thewilsonator** [comment]
For a small PR like this it's probably easier to just recreate it locally on stable and then and open a new PR. Saves messing around with git. Just make sure to select stable when choosing with branch to target.

**Comment by bangbangsheshotmedown** [comment]
> For a small PR like this it's probably easier to just recreate it locally on stable and then and open a new PR. Saves messing around with git. Just make sure to select stable when choosing with branch to target.

Do you think it is worth it? I checked stable and that code lives in a different file, I feel like it'll cause some issues when merging later?

**Comment by thewilsonator** [comment]
If you are fine waiting a while for it to get into a release then you don't need to. Do you want me to merge this as is now?

**Comment by bangbangsheshotmedown** [comment]
> If you are fine waiting a while for it to get into a release then you don't need to. Do you want me to merge this as is now?

I use LDC for release build, so it doesn't really impact me, I guess you can merge it as is

**Pull request merged**